### PR TITLE
Fix #1304 Implemented WMTS on OpenLayers

### LINKED ISSUE #1304: WMTS support for OL3

### PR DESCRIPTION
**Pull request by sp13adt**
#1304 Implemented support for WMTS on OpenLayers.

### CODE REVIEW

File: web/client/config.json [hunk 1]
```diff
@@ -102,6 +102,25 @@
         }
 			},
       {
+        "type": "wmts",
```
Comment (mbarto) [inline web/client/config.json]: Please remove this layer from default configuration

File: web/client/components/map/openlayers/plugins/WMTSLayer.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+function getWMSURLs( urls ) {
+    return urls.map((url) => url.split("\?")[0]);
+}
+
+Layers.registerType('wmts', {
+    create: (options) => {
+        const urls = getWMSURLs(isArray(options.url) ? options.url : [options.url]);
+        const queryParameters = wmtsToOpenlayersOptions(options) || {};
+        var projection = ol.proj.get('EPSG:900913');
```
Comment (mbarto) [inline web/client/components/map/openlayers/plugins/WMTSLayer.js]: We should use the srs in options, not a fixed 900913

File: web/client/components/map/openlayers/plugins/WMTSLayer.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        }
+        return new ol.layer.Tile({
+            opacity: options.opacity !== undefined ? options.opacity : 1,
+            source: new ol.source.WMTS(objectAssign({
+              urls: urls,
+              layer: options.name,
+              matrixSet: options.tileMatrixSet,
+              format: 'image/png',
+              tileGrid: new ol.tilegrid.WMTS({
+                origin: ol.extent.getTopLeft(projectionExtent),
```
Comment (mbarto) [inline web/client/components/map/openlayers/plugins/WMTSLayer.js]: We should consider using originX and originY options as we do for leaflet

File: web/client/components/map/openlayers/plugins/WMTSLayer.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+Layers.registerType('wmts', {
+    create: (options) => {
+        const urls = getWMSURLs(isArray(options.url) ? options.url : [options.url]);
+        const queryParameters = wmtsToOpenlayersOptions(options) || {};
+        urls.forEach(url => SecurityUtils.addAuthenticationParameter(url, queryParameters));
+        let matrixIds = new Array(22);
+        for (let z = 0; z < 22; ++z) {
+        // generate matrixIds arrays for this WMTS
+            matrixIds[z] = options.tileMatrixPrefix + z;
```
Comment (offtherailz) [inline web/client/components/map/openlayers/plugins/WMTSLayer.js]: We should have a list of matrixIds as a configuration option. In the future it have to be populated checking the catalog. If the list is not present in configuration, this autogeneration method can be the solution.

File: web/client/components/map/openlayers/plugins/WMTSLayer.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+function getWMSURLs( urls ) {
+    return urls.map((url) => url.split("\?")[0]);
+}
+
+Layers.registerType('wmts', {
+    create: (options) => {
+        const urls = getWMSURLs(isArray(options.url) ? options.url : [options.url]);
+        const queryParameters = wmtsToOpenlayersOptions(options) || {};
+        urls.forEach(url => SecurityUtils.addAuthenticationParameter(url, queryParameters));
+        let matrixIds = new Array(22);
```
Comment (offtherailz) [inline web/client/components/map/openlayers/plugins/WMTSLayer.js]: Tile matrix zoom levels are not 22 everywhere. Even when auto-generation is used, the list of resolutions and matrixIds can be variable. In general we should follow the information from WMTS. 
Keep in mind these 2 examples: 
 * http://basemap.nationalmap.gov/arcgis/rest/services/USGSHydroNHD/MapServer/WMTS/1.0.0/WMTSCapabilities.xml
 * http://basemap.nationalmap.gov/arcgis/rest/services/USGSHydroNHD/MapServer/WMTS/1.0.0/WMTSCapabilities.xml

As you see the identifier and the list are not the same.

**Comment by mbarto** [comment]
Replaced by #1397

**Pull request closed without merging**